### PR TITLE
feat(guard): add kim command safety extension

### PR DESCRIPTION
**Pull request by pratikwayal01**
Adds an external opt-in \command.kim\ command safety extension that reviews \kim add\, \kim remove\, \kim update\, and \kim stop\ while leaving the read-only \kim list\, \kim status\, and \kim logs\ commands automatic.

- New detector module \command_kim_extensions.py\ with four \
eview\ rules and one extension spec.
- Registered in \command_builtin_extension_catalog.py\, marked \external\ in the trust-class map, and published as \contributions/extensions/command.kim.json\.
- Added \	ests/test_guard_command_kim_extensions.py\ verifying mutations reach review observations and read-only commands stay untouched.
- Regenerated the extension directory (\docs/guard/extensions/README.md\), public catalog (\catalog.v1.json\), and the extension-control baseline fixture.

Existing unrelated failure: \	est_github_pr_create_body_file_honors_proposal_permission_toggle\ fails identically on clean \main\ at HEAD.

## Summary by CodeRabbit

* **New Features**
  * Expanded review coverage for potentially destructive Kim commands affecting reminder configuration, daemon state, and notification settings.
  * Added coverage for enabling, disabling, importing, exporting, editing, starting, stopping, updating, and uninstalling.
  * Read-only Kim commands, including logs, validation, shell completion, and non-effectful sound, Slack, and export forms, remain automatic.
  * Added the Kim command extension to the external, opt-in protection catalog.

* **Documentation**
  * Updated documented Kim command coverage from 4 to 16 rules.

### CODE REVIEW

**Review by kantorcodes — DISMISSED** [review]
Nice start. One blocker before we can take this: the extension only reviews `add/remove/update/stop`, but the current KIM CLI has other state-changing paths that would still run automatically, including `enable`, `disable`, `start`, `remind`, `import`, `self-update` and `uninstall` (plus the mutating sound/config paths). That makes the current “mutate config or stop the daemon; read-only inspection stays automatic” boundary too broad.

Please either cover the rest of the effectful surface, or narrow the extension metadata/scope so uncovered mutations aren’t implicitly treated as safe. Add tests for whichever boundary you choose.

**Comment by pratikwayal01** [comment]
Addressed the three review findings and expanded coverage to the full mutating surface.

**Qodo finding 1 (staging artifacts):** `command.kim.json` is now staged by `stage_guard_cloud_review_artifacts.py`, so the native builder archive carries the kim contribution alongside the other external extensions.

**Qodo finding 2 (empty executables/ecosystem IDs):** the kim spec now declares `executables=(kim,)` and `ecosystem_ids=(kim,)`, so the exported catalog entry carries `executables: [kim]` (previously `[]`). This also surfaces coverage of the `python -m kim` / `python3 -m kim` / `py -m kim` / `exec` / `xargs` launcher matrix.

**Qodo finding 3 (launcher coverage):** the extension now reviews every effectful kim subcommand across the same launcher matrix used by repo2nb, with wrapper leading options (`-n -P -I -L -s`) for exec/xargs and fail-secure unknown-option handling.

**Expanded surface (was 4 rules, now 16):** add, remove, update, enable, disable, remind (one-shot), import, export, start, stop, edit, sound, slack, interactive (+`-i` alias), self-update, uninstall. Read-only `list`/`status`/`logs`/`validate`/`completion` and root `-v` stay automatic; the daemon-internal `_remind-fire` subcommand remains excluded (it fires every scheduled notification, not a user-proposed command).

Tests extended in `tests/test_guard_command_kim_extensions.py`, global rule count updated (236 -> 248), catalog/baseline fixtures and docs regenerated. Note: 3 pre-existing failures in `test_guard_command_extensions.py::test_github_pr_create_body_file_honors_proposal_permission_toggle` reproduce on `main` without these changes.

File: src/codex_plugin_scanner/guard/runtime/command_kim_extensions.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+KIM_COMMAND_EXTENSION_SPECS = (
+    CommandExtensionSpec(
+        extension_id="command.kim",
+        name="kim reminder protection",
+        description=(
+            "Reviews kim commands that mutate the local reminder config or "
+            "stop the reminder daemon; read-only inspection stays automatic."
+        ),
+        action_classes=(
```
[the remark this answers is not in the document]
Comment (pratikwayal01) [reply src/codex_plugin_scanner/guard/runtime/command_kim_extensions.py:438]: Addressed in 162fe0976 (pushed to the PR branch). `KIM_ACTION_RISK_CLASSES` now registers all 16 kim action classes (each resolving to `destructive_shell`) and is wired into `COMMAND_ACTION_RISK_CLASSES` in `command_action_risk_classes.py`. `test_kim_action_classes_map_to_runtime_risk_classes` asserts `risk_classes_for_command_action` parity for every entry.

File: contributions/extensions/command.kim.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  ],
+  "saferAlternatives": [
+    "Preview the reminder list and daemon state with kim list and kim status before mutating them.",
+    "Confirm the exact reminder name before removing or updating it."
+  ],
+  "detector": {
+    "kind": "python-module",
+    "module": "codex_plugin_scanner.guard.runtime.command_kim_extensions"
+  }
+}
```
[the remark this answers is not in the document]
Comment (pratikwayal01) [reply contributions/extensions/command.kim.json]: Fixed in 162fe0976 - `contributions/extensions/command.kim.json` now ends with a trailing newline.

File: tests/test_guard_command_kim_extensions.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        assert (expected_rule_id, expected_action_class) in matched, command
+
+    for command in KIM_READONLY_COMMANDS:
+        observations = BUILT_IN_COMMAND_EXTENSION_REGISTRY.observations(
+            parse_shell_command(command, cwd=tmp_path, home_dir=tmp_path)
+        )
+        assert all(item.extension.extension_id != "command.kim" for item in observations), command
+
+
+KIM_READONLY_COMMANDS: tuple[str, ...] = (
```
[the remark this answers is not in the document]
Comment (pratikwayal01) [reply tests/test_guard_command_kim_extensions.py]: Fixed in 162fe0976 - `KIM_READONLY_COMMANDS` is now defined above `test_kim_mutations_reach_review_and_readonly_commands_stay_safe`, consistent with `KIM_REVIEW_CASES`.

File: src/codex_plugin_scanner/guard/runtime/command_kim_extensions.py [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+_KIM_ENABLE = _kim_review_matcher("enable")
+_KIM_DISABLE = _kim_review_matcher("disable")
+_KIM_REMIND = _kim_review_matcher("remind", options_with_values=_KIM_REMIND_OPTIONS_WITH_VALUES)
+_KIM_IMPORT = _kim_review_matcher("import", options_with_values=_KIM_IMPORT_OPTIONS_WITH_VALUES)
+_KIM_EXPORT = _kim_review_matcher("export", options_with_values=_KIM_EXPORT_OPTIONS_WITH_VALUES)
+_KIM_START = _kim_review_matcher("start")
+_KIM_STOP = _kim_review_matcher("stop")
+_KIM_EDIT = _kim_review_matcher("edit")
+_KIM_SOUND = _kim_review_matcher("sound", options_with_values=_KIM_SOUND_OPTIONS_WITH_VALUES)
+_KIM_SLACK = _kim_review_matcher("slack", options_with_values=_KIM_SLACK_OPTIONS_WITH_VALUES)
```
[the remark this answers is not in the document]
Comment (pratikwayal01) [reply src/codex_plugin_scanner/guard/runtime/command_kim_extensions.py]: Addressed in 162fe0976 (pushed to the PR branch). `_KIM_SOUND` and `_KIM_SLACK` now require an effectful flag (`--set`/`--clear`/`--test`/`--enable`/`--disable` and `--test`, matching kim's CLI), so bare `kim sound`/`kim slack` config-inspection forms stay automatic. Regression cases updated in `test_guard_command_kim_extensions.py`; stdout-only `kim export` without `-o`/`--output` is kept automatic for the same reason.

**Comment by pratikwayal01** [comment]
Thanks for the reviews. Commit `162fe0976` (pushed) addresses the remaining actionable comments:

**CodeRabbit / Kilo**
- `kim sound` / `kim slack` bare forms now stay automatic: `_KIM_SOUND` and `_KIM_SLACK` require an effectful flag (`--set`/`--clear`/`--test`/`--enable`/`--disable`, and `--test`, matching kim's CLI). stdout-only `kim export` without `-o`/`--output` is also kept automatic. Regression cases updated.
- Kilo: all 16 kim action classes are now registered via `KIM_ACTION_RISK_CLASSES` and spliced into `COMMAND_ACTION_RISK_CLASSES`; a parity test covers `risk_classes_for_command_action`.
- Kilo: `contributions/extensions/command.kim.json` now ends with a trailing newline.
- Kilo: `KIM_READONLY_COMMANDS` moved above the test that uses it.

**Qodo** (these three snapshots predate the latest commits)
- "Packaged builds lose kim branding" - already fixed by the `_ARTIFACTS` staging map in `4da2716f5`.
- "Runtime catalogs hide the kim tool" - already fixed by `executables=("kim",)` / `ecosystem_ids=("kim",)` in `b8b4870b4`.
- "Batch reminder removals bypass review" - `xargs -n1 kim remove` (and `-n 1`, `--`, piped variants) is already matched by the launcher matrix shipped in `b8b4870b4`; verified locally for all listed forms.

Not in scope for this PR: the CodeRabbit docstring metric (informational) and Kilo CLI-semantics/manifest-tracking items, which are tracked separately.

Test status: kim + extension tests, catalog detail, directory parity, trust, and builder native/repository all green; the only failures in `test_guard_command_extensions.py` are the three pre-existing GitHub proposal-toggle failures I verified also fail on `origin/main`.

**Review by kantorcodes — DISMISSED** [review]
looks good now. the expanded command coverage and read-only variants address my blocker.

**Comment by pratikwayal01** [comment]
Fixed the failing CI check (tests/test_policy_bundle_delivery_runtime.py::test_signed_cloud_extension_projection_matches_shared_vector): regenerated contracts/managed-controls/v1/extension-projection-digest-vector.json + the matching test constants for the new build_builtin_extension_catalog_wire() digest after the kim command extensions were added (commit 1427164). Verified locally — 49/49 related tests pass.

@kantorcodes — the new CI runs for this head are blocked on first-time-contributor approval. Could you approve the run (or enable runs for the PR)? Thanks.

**Review by kantorcodes — COMMENTED** [review]
I rechecked the current Kim head and the prior substantive findings are resolved: packaging metadata is staged, runtime identity fields are populated, xargs coverage was added, effectful `sound`/`slack` forms were tightened, and action-risk registration is present. I do not see a remaining code-level blocker in the active review threads.

I am not approving this exact head because it is currently non-mergeable against `main`. Please rebase/merge current `main`, resolve the generated catalog/baseline conflicts, regenerate those artifacts from the resolved tree, and rerun the focused Kim/extension/registry/trust/package checks. If the rebased head stays semantically equivalent and green, this should be approval-ready on the next pass.

**Review by kantorcodes — COMMENTED** [review]
Re-reviewed the rebased head `fca04c2ee50789974f8de839be704102babd98ff`. The prior mergeability/rebase blocker is cleared, the Kim-specific matcher/metadata/generated-artifact changes remain in good shape, and I do not see a new code-level blocker from the rebase itself.

I am not approving or merging this exact head yet because the required fork Actions are still in GitHub's `action_required` state rather than completed/green. That is a workflow-authorization gate, not a code failure. Once the required CI/security/Extension-builder/installed-artifact jobs actually run and pass on this SHA, this is ready for the approval/merge step from my side.

**Review by kantorcodes — COMMENTED** [review]
`main` moved again. Please merge current `main` (`b910d77`) and regenerate the derived catalog/baseline/projection artifacts before the next pass; this repo requires branches to be current before merge.
